apps/emqx_authz/src/emqx_authz_http.erl

```diff
@@ -121,30 +141,65 @@ serialize_body(<<"application/json">>, Body) ->
 serialize_body(<<"application/x-www-form-urlencoded">>, Body) ->
     query_string(Body).
 
-replvar(Str0, PubSub, Topic,
-        #{username := Username,
-          clientid := Clientid,
-          peerhost := IpAddress,
-          protocol := Protocol,
-          mountpoint := Mountpoint
-         }) when is_list(Str0);
-                 is_binary(Str0) ->
+
```

Comment:
All these functions (replacing variables and traversing structs) will are to be refactored and put into a single templating module.

Reply:
Currently, if I understood correctly, it still may interpolate twice, right?